src/main/kotlin/org/jetbrains/research/testspark/actions/TestSparkAction.kt

```diff
@@ -74,14 +74,18 @@ class TestSparkAction : AnAction() {
     /**
      * Updates the state of the action based on the provided event.
      *
-     * @param e the AnActionEvent object representing the event
+     * @param e `AnActionEvent` object representing the event
      */
     override fun update(e: AnActionEvent) {
-        val file = e.dataContext.getData(CommonDataKeys.PSI_FILE)!!
-        val psiHelper = PsiHelperProvider.getPsiHelper(file)
-        if (psiHelper == null) {
-            // TODO exception
+        val file = e.dataContext.getData(CommonDataKeys.PSI_FILE)
+
+        if (file == null) {
+            e.presentation.isEnabledAndVisible = false
+            return
         }
+
+        val psiHelper = PsiHelperProvider.getPsiHelper(file)
+        e.presentation.isEnabledAndVisible = (psiHelper != null) && psiHelper.availableForGeneration(e)
```

Comment:
--> `psiHelper?.availableForGeneration(e) ?: false`

Reply:
Thought about it, but I do not like the elvis operation and `?` used in the single expression. Also, it is possible to change it to `psiHelper?.availableForGeneration(e) === true`, but I do not like it either.

Overall, it is just a style preference, and I prefer the current variant.